extensions/music/src/play-library-album.tsx

```diff
@@ -0,0 +1,71 @@
+import { ActionPanel, closeMainWindow, List, showToast, ToastStyle, useNavigation } from "@raycast/api";
+import { isLeft } from "fp-ts/lib/Either";
+import React, { useState } from "react";
+import { playAlbum, seachForAlbum } from "./util/controls";
+import { Album } from "./util/models";
+import { parseResult } from "./util/parser";
+
+export default function PlayLibraryAlbum() {
+  const [albums, setAlbums] = useState<Album[] | null>();
+  const { pop } = useNavigation();
+
+  const onSearch = async (next: string) => {
+    setAlbums(null);
+    if (!next || next?.length < 1) {
+      setAlbums([]);
+      return;
+    }
+    const raw = await seachForAlbum(next)();
```

Comment:
I would write this in a more functional way, following `fp-ts` docs. 
But even like this is ok for me, I can't force you to write code in a weird way 😄 

Something like this should be fine.. 👀  (check types)
```ts
const result: Album[] = await pipe(
	next,
	searchForAlbum,
	TE.matchW(
		() => showToast(ToastStyle.Failure, 'Could not get albums'),
		albums => pipe(
			albums,
			O.fromNullable,
			O.matchW(
				() => [],
				A.map(album => parseResult<Album>(album))
			)
		)
	),
)()
```

Reply:
If you need any help or have some doubts you can contact me on Slack